exchanges/okx/okx_test.go

```diff
@@ -6093,6 +6094,46 @@ func TestGenerateSubscriptions(t *testing.T) {
 	testsubs.EqualLists(t, exp, subs)
 }
 
+func TestWsCandles(t *testing.T) {
+	t.Parallel()
+	e := new(Exchange) //nolint:govet // Intentional shadow
+	require.NoError(t, testexch.Setup(e), "Setup must not error")
+
+	err := e.WsConnectBusiness(t.Context())
+	require.NoError(t, err, "WsBusinessConnect must not error")
+
+	err = e.BusinessSubscribe(t.Context(), subscription.List{{Channel: channelCandle1D}})
+	require.ErrorIs(t, err, currency.ErrCurrencyPairEmpty)
+
+	p := currency.Pairs{
+		mainPair,
+		currency.NewPairWithDelimiter("ETH", "USDT", "-"),
+		currency.NewPairWithDelimiter("OKB", "USDT", "-"),
+	}
+
+	for i, ch := range []string{channelCandle1D, channelMarkPriceCandle1M, channelIndexCandle1H} {
+		err := e.BusinessSubscribe(t.Context(), subscription.List{{Channel: ch, Pairs: p[i : i+1]}})
+		require.NoErrorf(t, err, "BusinessSubscribe %s-%s must not error", ch, p[i])
+	}
+
+	var got currency.Pairs
+	assert.Eventually(t, func() bool {
+		select {
+		case a := <-e.Websocket.DataHandler:
+			switch v := a.(type) {
+			case websocket.KlineData:
+				got = got.Add(v.Pair)
+			case []CandlestickMarkPrice:
+				if len(v) > 0 {
+					got = got.Add(v[0].Pair)
+				}
+			}
+		default:
+		}
+		return len(got) == 3
```

Comment:
[nitpick] I think the thing that would make me happiest is to make it totally unambiguous about what you're expecting here. 
```go
return got.Contains(mainPair, true) &&
			got.Contains(currency.NewPairWithDelimiter("ETH", "USDT", "-"), true) &&
			got.Contains(currency.NewPairWithDelimiter("OKB", "USDT", "-"), true)
	
```
Or making `got` `got := make(map[currency.Pair]struct{})` to continue using the len check

Reply:
`pairs.Add` [ignores duplicates](https://github.com/thrasher-corp/gocryptotrader/blob/master/currency/pairs.go#L219)

Do you think we need a comment to explain it?
Normally I'd just add one, but it feels like once we're past this review, the comment would be self-evident

Reply:
I swear I checked that, but I'll look at what you've written and if so it should be fine. I modified your test have len == 5 match, which is what made me make my comment. I'll recheck before guiding what to do

Reply:
`len == 5` should have failed. It fails for me.
I don't see how it could pass ... We don't connect any other websocket, or any other subscriptions.
It seems impossible you could get 5. If you can reproduce I'll be very interested.